Unix/deprecated/httpclientcxx/httpclientcxx.cpp

```diff
@@ -726,6 +731,22 @@ Result HttpClient::Connect(
     return httpclient::OKAY;
 }
 
+void HttpClient::Cancel( void )
+{
+    pthread_mutex_lock(&s_deleteSemaphoreLock);
```

Comment:
Well, I didn't design this code, this is ported from 1.0.6.5, and I don't have knowledge into how NotifyItem and PostItem work at all. But, it appears we do assume that this will get called only after connect is called. I believe StartRequest also makes the same assumption, that it should get called only after a Connect is called (It too also uses semaphores and mutexes that are initialized in Connect).